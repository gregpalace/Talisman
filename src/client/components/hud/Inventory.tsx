import React from 'react';
import { connect } from 'react-redux';
import EmptySlot from './emptySlot';

import { IInventory } from '../../../typings/HUDTypes';
import * as types from '../../constants/actionTypes';

import '../../../stylesheets/HUD-Styles/playerInventory.scss';

const Inventory = ({ hp, mp, inventorySlots }: IInventory) => {
  const slots = new Array(types.INVENTORY_SLOTS).fill(null);
  const itemSlots = slots.map((el) => {
    el === null;
  });

  return (
    <div id='player-inventory'>
      <ul id='player-table'>
        <li>Level: 5</li>
        <li>HP: {hp}</li>
        <li>MP: {mp}</li>
      </ul>
      <aside id='inventory-items'>
        <section className='items'>
          <EmptySlot>{itemSlots[0]}</EmptySlot>
          <EmptySlot>{itemSlots[1]}</EmptySlot>
          <EmptySlot>{itemSlots[2]}</EmptySlot>
        </section>
        <section className='items'>
          <EmptySlot>{itemSlots[4]}</EmptySlot>
          <EmptySlot>{itemSlots[5]}</EmptySlot>
          <EmptySlot>{itemSlots[6]}</EmptySlot>
        </section>
      </aside>
      <hr />
      <footer className='inventory-slots'>
        <h4 className='slots'>Slots Available: {inventorySlots}</h4>
      </footer>
    </div>
  );
};

export default Inventory;
