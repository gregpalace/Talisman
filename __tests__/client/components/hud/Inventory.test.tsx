import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Inventory from '../../../../src/client/components/hud/Inventory';
import EmptySlot from '../../../../src/client/components/hud/emptySlot';

describe('React unit tests', () => {
  let wrapper: any;

  const props = {
    hp: 130,
    mp: 230,
    inventorySlots: 1,
  };

  beforeEach(() => {
    wrapper = shallow(<Inventory {...props} />);
  });

  describe('<Inventory />', () => {
    xit('should render properly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render our players main inventory', () => {
      expect(wrapper.find('div').prop('id')).toBe('player-inventory');
    });

    it('should render a list of player stats', () => {
      const ul = wrapper.find('ul');
      expect(ul.prop('id')).toBe('player-table');
      ul.find('li').forEach((li: any, index: number) => {
        if (index === 0) {
          expect(li.text()).toBe('Level: 5');
        } else if (index === 1) {
          expect(li.text()).toBe('HP: 130');
        } else if (index === 2) {
          expect(li.text()).toBe('MP: 230');
        }
      });
    });

    it('should render an aside full of inventory items', () => {
      const aside = wrapper.find('aside');
      expect(aside).toBeDefined();
      expect(aside.prop('id')).toBe('inventory-items');
    });

    it('should render <EmptySlot /> components for a players inventory', () => {
      expect(wrapper.find(EmptySlot)).toBeDefined();
      expect(wrapper.find(EmptySlot).length).toEqual(6);
    });

    it('should render a <footer> containing inventory count', () => {
      const footer = wrapper.find('footer');
      expect(footer).toBeDefined();
      expect(footer.prop('className')).toBe('inventory-slots');
      expect(footer.find('h4'));
      expect(footer.find('h4').text()).toBe('Slots Available: 1');
    });
  });
});
