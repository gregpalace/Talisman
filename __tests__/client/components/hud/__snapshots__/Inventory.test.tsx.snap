// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests <Inventory /> should render properly 1`] = `
<div
  id="player-inventory"
>
  <ul
    id="player-table"
  >
    <li>
      Level: 5
    </li>
    <li>
      HP: 
      130
    </li>
    <li>
      MP: 
      230
    </li>
  </ul>
  <aside
    id="inventory-items"
  >
    <section
      className="items"
    >
      <EmptySlot />
      <EmptySlot />
      <EmptySlot />
    </section>
    <section
      className="items"
    >
      <EmptySlot />
      <EmptySlot />
      <EmptySlot />
    </section>
  </aside>
  <hr />
  <footer
    className="inventory-slots"
  >
    <h4
      className="slots"
    >
      Slots Available: 
      1
    </h4>
  </footer>
</div>
`;
